### PR TITLE
Google Lens OCR Updates

### PR DESCRIPTION
**Pull request by AgnesRiber**
Improvements:

1. rotation of user-agents for each request.
2. a new text log to save the content.

### CODE REVIEW

**Comment by HIllya51** [comment]
These modifications seem meaningless.

File: py/LunaTranslator/ocrengines/googlelens.py [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+        "Mozilla/5.0 (Linux; Tizen 2.3; SmartHub; SMART-TV; SmartTV; U; Maple2012) waipu/2023.6.11-2494 (TV; Samsung; 17_KANTM_UHD; waipu; Tizen 3.0)"
+    ]
 
     def ocr(self, imagebinary):
-
         regex = re.compile(r">AF_initDataCallback\(({key: 'ds:1'.*?)\);</script>")
-
         timestamp = int(time.time() * 1000)
-        url = "https://lens.google.com/v3/upload?stcs={}".format(timestamp)
+        url = f"https://lens.google.com/v3/upload?stcs={timestamp}"
```
Comment (HIllya51) [inline py/LunaTranslator/ocrengines/googlelens.py:17]: don't change this

File: py/LunaTranslator/ocrengines/googlelens.py [hunk 2]
```diff
@@ -21,13 +26,17 @@ def ocr(self, imagebinary):
...
-
+        
         text = lens_object["data"][3][4][0]
-        return text[0] if text else None
+        
+        if text:
+            text_list = list(text[0]) if text else []
+            with open("log.txt", "a", encoding="utf-8-sig") as file:
+                file.write(" ".join(text_list) + "\n")
+        return text_list if text else None
```
Comment (HIllya51) [inline py/LunaTranslator/ocrengines/googlelens.py:42]: don't write this.

**Comment by HIllya51** [comment]
For my testing, these modifications are meaningless.

**Comment by HIllya51** [comment]
I use VPN to access Google Lens. After my testing, result maybe related to the IP of the VPN. Different IPs in the same country also have different results. Using different UAs has no impact on the results.

**Comment by AgnesRiber** [comment]
> I use VPN to access Google Lens. After my testing, result maybe related to the IP of the VPN. Different IPs in the same country also have different results. Using different UAs has no impact on the results.

I see... Proxy rotation don't work?

**Comment by HIllya51** [comment]
It has no effect.

**Pull request closed without merging**